src/test/java/teammates/sqllogic/core/AccountsLogicTest.java

```diff
@@ -0,0 +1,170 @@
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.NotificationStyle;
+import teammates.common.datatransfer.NotificationTargetUser;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.storage.sqlapi.AccountsDb;
+import teammates.storage.sqlentity.Account;
+import teammates.storage.sqlentity.Notification;
+import teammates.storage.sqlentity.ReadNotification;
+import teammates.test.BaseTestCase;
+
+/**
+ * SUT: {@link AccountsLogic}.
+ */
+public class AccountsLogicTest extends BaseTestCase {
+
+    private AccountsLogic accountsLogic = AccountsLogic.inst();
+
+    private AccountsDb accountsDb;
+
+    private NotificationsLogic notificationsLogic;
+
+    @BeforeMethod
+    public void setUpMethod() {
+        accountsDb = mock(AccountsDb.class);
+        notificationsLogic = mock(NotificationsLogic.class);
+        accountsLogic.initLogicDependencies(accountsDb, notificationsLogic);
+    }
+
+    @Test
+    public void testUpdateReadNotifications_success() throws InvalidParametersException, EntityDoesNotExistException {
+        Account account = generateTypicalAccount();
+        Notification notification = generateTypicalNotification();
+        String googleId = account.getGoogleId();
+        UUID notificationId = notification.getId();
+
+        when(accountsDb.getAccountByGoogleId(googleId)).thenReturn(account);
+        when(notificationsLogic.getNotification(notificationId)).thenReturn(notification);
+
+        List<UUID> readNotificationIds =
+                accountsLogic.updateReadNotifications(googleId, notificationId, notification.getEndTime());
+
+        verify(accountsDb, times(1)).getAccountByGoogleId(googleId);
+        verify(notificationsLogic, times(1)).getNotification(notificationId);
+
+        assertEquals(1, readNotificationIds.size());
+        assertEquals(notificationId, readNotificationIds.get(0));
+
+        List<ReadNotification> accountReadNotifications = account.getReadNotifications();
+        assertEquals(1, accountReadNotifications.size());
+        ReadNotification readNotification = accountReadNotifications.get(0);
+        assertEquals(account, readNotification.getAccount());
+        assertEquals(notification, readNotification.getNotification());
+
+        List<ReadNotification> notificationReadNotifications = notification.getReadNotifications();
+        assertEquals(1, notificationReadNotifications.size());
+        readNotification = notificationReadNotifications.get(0);
+        assertEquals(account, readNotification.getAccount());
+        assertEquals(notification, readNotification.getNotification());
```

Comment:
We have no real use case for getting ReadNotifications from notifications. Maybe should remove this getter and setter entirely.

Reply:
In fact, I think we can remove the association on the notification side entirely. This would simplify a lot of the code.